streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -93,10 +115,8 @@ public void shouldProcessFromSinkTopic() {
 
         source.process(processorSupplier);
 
-        driver.setUp(builder);
-        driver.setTime(0L);
-
-        driver.process("topic-source", "A", "aa");
+        driver = new TopologyTestDriver(builder.internalTopologyBuilder, props);
```

Comment:
Why can't we pass in `builder.build()` ?

Reply:
Here `.build()` returns a `ProcessorTopology` instead of a `Topology`, which the constructor expects.

Reply:
Why do we need this? `TopologyTestDriver` has

```
    public TopologyTestDriver(final Topology topology,
                              final Properties config) {
```

thus we should be able to pass in the `Topology` -- what do I miss?

Reply:
The problem is that `KStreamBuilder` extends `org.apache.kafka.streams.processor.TopologyBuilder`, so it's `.build()` method returns a `ProcessorTopology` instead of a `Topology`. Consequently, we face the same problem as in the tests for `TopologyBuilder`. I got around it by using the wrapper class which exposes the protected constructor, as you suggested above.